### ISSUE TITLE
BW::Location CLLocationManagerDelegate method deprecation

### ISSUE THREAD

**Issue by cognitiveflux** [rubymotion/BubbleWrap#421]
The `CLLocationManagerDelegate` method `locationManager:didUpdateToLocation:fromLocation` has been deprecated since two prior major version of iOS and recommends `locationManager:didUpdateLocations` instead.

While the new delegate method guarantees at least one `CLLocation` object in the response, additional objects may or may not be included.

I can create a branch and implement the new delegate method, however I created this issue to collect feedback on how it should be implemented because a `result[:from]` value is no longer guaranteed as provided in this [Callback](https://github.com/rubymotion/BubbleWrap/blob/master/motion/location/location.rb#L181).

A possible implementation would require `result[:from]` to be removed and replaced with an *optional* `result[:previous_locations]`, for example.  Alternatively, the callback could store the previous location in an instance variable and return it for `result[:from]` in the event that the `locations` array returned by the delegate method only has one member, thus preserving the `BW::Location` functionality.  That being said, in the current implementation `result[:from]` will ways be `nil` on the first location update.

**Comment by Siggs2000** [comment]
Hey everyone,

I was just checking in to see if there was any solution to this that I've missed. I notice that I get location errors on simulator pretty often unless I rake:clean and rebuild all the time. 

It works often but not all the time... randomly...

I'll end up with this error: 

```
undefined method `coordinate' for {...}:Hash (NoMethodError)
```

**Comment by cognitiveflux** [comment]
Are you using a branch with #422?  I retried several times to run the tests but running tests against `master` failed and I didn't have the time to figure out what other commit was causing the tests to fail before trying to test with my branch.  I ended up implementing location services without BW.

Separate issue, I know that if the simulator is left running (~ 15 minutes) it stops responding to location requests, whether using BW+422 or implementing `CLLocationManagerDelegate`, so at least that much I would attribute as an issue directly with the Simulator because I have never come across it on a device.
